### PR TITLE
refactor: use registration hook

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- Changes are mostly improving types, cleanups, setting the right parameters for the required traits, some minor bug fix in logout, etc.
- The registration process will be used in different places, I created a hook to the registration auth service for easier use.
- I have also centralized the processing of data for both social provider and password method.
- Would love to hear feedback on areas we can improve 🚢 

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### CODE REVIEW

File: packages/shared/src/lib/auth.ts [hunk 1]
```diff
@@ -97,18 +133,16 @@ export interface AuthSession extends LogoutSessionData {
...
-  created_at: '2022-07-21T05:18:27.975693Z';
-  updated_at: '2022-07-21T05:18:27.975693Z';
-  refresh: false;
-  requested_aal: 'aal1';
+export interface SuccessfulRegistrationData {
+  session: AuthSession;
+  identity: Identity;
 }
 
+const authUrl = 'http://127.0.0.1:4433';
```
Comment (rebelchris) [inline packages/shared/src/lib/auth.ts:141]: Probably best to put this in a env actually.

**Review by rebelchris — APPROVED** [review]
Nice, thanks a lot for going through these.
I think we can still cleanup some of the usecases, but this start will already really help.

Perhaps we can go over some things in a realtime session at one point.
(non blocking for this)

File: packages/shared/src/lib/auth.ts (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/lib/auth.ts:141]: Totally agree. Let's just make this value a fallback if the env is empty. Thank you!

**Comment by sshanzel** [comment]
@rebelchris totally agree. At some point we should do a sync session. Thank you for the review! 🚢

**Pull request merged**